Simplenote/src/main/java/com/automattic/simplenote/authentication/MagicLinkableFragment.kt

```diff
@@ -0,0 +1,105 @@
+package com.automattic.simplenote.authentication
+
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.Patterns
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.EditText
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.Fragment
+import com.automattic.simplenote.R
+import com.google.android.material.textfield.TextInputLayout
+import com.simperium.android.ProgressDialogFragment
+
+/**
+ * Base class used to share logic between sign up and login, specifically related to magic links.
+ */
+abstract class MagicLinkableFragment : Fragment() {
```

Comment:
Are you thinking on implementing more fragments related to magic links?

Reply:
Sorta. This class was based off of `SignupFragment`, which is similar. Both Sign Up and Sign In with magic link are similar. So it made sense to use that logic and share it between them. Additionally, we can likely remove the okhttp use from `SignupFragment` and move it over to a viewmodel. Hence why I created the `SimpleHttp` helper object. But I didn't want to inflate the work here further.